src/wallet/did_wallet/did_wallet.py

```diff
@@ -0,0 +1,760 @@
+        innersol = Program.to(
+            [
+                2,
+                coin.amount,
+                puzhash,
+                coin.name(),
+                coin.puzzle_hash,
+                parent_innerpuzhash_amounts_for_recovery_ids,
+            ]
+        )
+        # full solution is (parent_info my_amount solution)
+        innerpuz = self.did_info.current_inner
+        full_puzzle: str = did_wallet_puzzles.create_fullpuz(
+            innerpuz,
+            self.did_info.my_did,
+        )
+        parent_info = await self.get_parent_for_coin(coin)
+        fullsol = Program.to(
+            [
+                [
+                    parent_info.parent_name,
+                    parent_info.inner_puzzle_hash,
+                    parent_info.amount,
+                ],
+                coin.amount,
+                innersol,
+            ]
+        )
+        list_of_solutions = [
+            CoinSolution(
+                coin,
+                clvm.to_sexp_f([full_puzzle, fullsol]),
+            )
+        ]
+        sigs = []
+        aggsig = AugSchemeMPL.aggregate(sigs)
+        if spend_bundle is None:
+            spend_bundle = SpendBundle(list_of_solutions, aggsig)
+        else:
+            spend_bundle = spend_bundle.aggregate(
+                [spend_bundle, SpendBundle(list_of_solutions, aggsig)]
+            )
+
+        did_record = TransactionRecord(
+            confirmed_at_index=uint32(0),
+            created_at_time=uint64(int(time.time())),
+            to_puzzle_hash=puzhash,
+            amount=uint64(coin.amount),
+            fee_amount=uint64(0),
+            incoming=False,
+            confirmed=False,
+            sent=uint32(0),
+            spend_bundle=spend_bundle,
+            additions=spend_bundle.additions(),
+            removals=spend_bundle.removals(),
+            wallet_id=self.wallet_info.id,
+            sent_to=[],
+            trade_id=None,
+        )
+        await self.standard_wallet.push_transaction(did_record)
+        return spend_bundle
+
+    async def get_new_innerpuz(self) -> Program:
+        devrec = await self.wallet_state_manager.get_unused_derivation_record(
+            self.standard_wallet.wallet_info.id
+        )
+        pubkey = bytes(devrec.pubkey)
+        innerpuz = did_wallet_puzzles.create_innerpuz(pubkey, self.did_info.backup_ids)
+
+        return innerpuz
+
+    async def get_new_inner_hash(self) -> bytes32:
+        innerpuz = await self.get_new_innerpuz()
+        return innerpuz.get_tree_hash()
+
+    async def get_innerhash_for_pubkey(self, pubkey: bytes):
+        innerpuz = did_wallet_puzzles.create_innerpuz(pubkey, self.did_info.backup_ids)
+        return innerpuz.get_tree_hash()
+
+    async def inner_puzzle_for_did_puzzle(self, did_hash: bytes32) -> Program:
+        record: DerivationRecord = await self.wallet_state_manager.puzzle_store.get_derivation_record_for_puzzle_hash(
+            did_hash.hex()
```

Comment:
Why are we passing in a `.hex()`? The name of the function suggests it wants a puzzle hash, not a hex string of the puzzle hash.